### DISCUSSION TITLE
Multi step form - let users delete their incomplete submission state?

### DISCUSSION THREAD

**Discussion by mandrasch** [verbb/formie#1593] [Q&A] [answered]
Hey, 

I wondered if there is already a quick way of letting users delete their (incomplete) submission?

E.g. if users are on a public computer - or if they just want to start over completely fresh with their form.

(My multi-step form is used by logged in users as well as guests (not logged in))

Thanks very much for advice!

Best regards,
Matthias

**Comment by engram-design** [comment, marked as answer]
We would probably need to implement a "Clear" or "Reset" button for forms to achieve this, but it's completely possible to DIY yourself with custom templates. We just didn't want to add too many buttons (Back, Save, Submit, Reset) to the buttons area for a page.

```twig
<a href="{{ actionUrl('formie/submissions/clear-submission', { handle: 'contactForm }) }}">Reset</a>
```

**Reply by mandrasch** [reply]
Hi @engram-design,

thanks very much for reply!

Added it to my twig template, but got 

```
verbb\formie\controllers\SubmissionsController::_getForm(): Argument #1 ($handle) must be of type string, null given, called in /var/www/html/vendor/verbb/formie/src/controllers/SubmissionsController.php on line 772
```

URL seemed to be fine `...index.php?p=actions/formie/submissions/clear-submission&handle=myForm` though.

**Reply by engram-design** [reply]
Fixed for the next release. To get this early, run `composer require verbb/formie:"dev-craft-4 as 2.0.38"`.

**Reply by mandrasch** [reply]
Awesome, thanks very much!  👍👍 :-)
